src/lib/libexceptions.js

```diff
@@ -283,16 +283,11 @@ var LibraryExceptions = {
   },
 #endif
 #if WASM_EXCEPTIONS
-  $getCppExceptionTag: () =>
-    // In static linking, tags are defined within the wasm module and are
-    // exported, whereas in dynamic linking, tags are defined in library.js in
-    // JS code and wasm modules import them.
-#if RELOCATABLE
-    ___cpp_exception // defined in library.js
-#else
-    wasmExports['__cpp_exception']
-#endif
-  ,
+  $getCppExceptionTag__deps: ['__cpp_exception'],
+  // In static linking, tags are defined within the wasm module and are
+  // exported, whereas in dynamic linking, tags are defined in libcore.js in
+  // JS code and wasm modules import them.
+  $getCppExceptionTag: () => ___cpp_exception,
```

Comment:
It looks like this splits the code from this file to two places, here and preamble.js. That seems less simple at first glance - are there other benefits?

Reply:
The removal of the dependency on `wasmExports` is the main benefit.   

That currently doesn't work in `WASM_ESM_INTEGRATION` mode.  We could make it work I suppose by adding an `import * as wasmExports` statement.. but it seems less precise that just importing what is used.

In the long run I think we should have a generic way to handle the non-function, non-global exports of a wasm module.

As of today we have special cases in a lot of places for `memory`, `__indirect_function_table` and `__cpp_exception` but I'd like to eliminate those if possible and replace them with generic export handling.

Reply:
Now I see, thanks.